repl/src/main/scala/ammonite/repl/interp/Interpreter.scala

```diff
@@ -135,12 +146,27 @@ class Interpreter(prompt0: Ref[String],
     s"Syntax Error: $msg\n$locationString"
   }
 
-  def processCorrectScript(parsedCode: Parsed.Success[Seq[Seq[String]]],
+  def processCorrectScript(rawParsedCode: Parsed.Success[Seq[(String, Seq[String])]],
                            evaluate: EvaluateCallback): Seq[ImportData] = {
 
+    var offset = 0
+    var parsedCode: mutable.ArrayBuffer[( String, Seq[String])] = mutable.ArrayBuffer()
+
+    // comment holds comments or empty lines above the code which is not caught along with code
+    for( (comment, code) <- rawParsedCode.get.value ){
+      val ncomment = comment + "\n"*offset
+
+      // 1 is added as Separator parser eats up the '\n' following @
+      offset = offset + comment.count(_ == '\n') + code.map(_.count(_ == '\n')).sum + 1
+      parsedCode = parsedCode :+ (ncomment, code)
```

Comment:
Since we're using `mutable.{ArrayBuffer, Buffer}` above, `:+` is O(n) time and thus the for-loop is O(n^2). Use `.append` instead which is O(1)